app/helpers/commodity_helper.rb

```diff
@@ -53,15 +53,7 @@ def applicable_excise_additional_codes
   end
 
   def applicable_document_codes
-    @applicable_document_codes ||=
-      {}.tap do |applicable_document_codes|
-        if user_session.deltas_applicable?
-          applicable_document_codes['uk'] = surface_document_codes(commodity: uk_filtered_commodity)
-          applicable_document_codes['xi'] = surface_document_codes(commodity: xi_filtered_commodity)
-        else
-          applicable_document_codes[user_session.commodity_source] = surface_document_codes
-        end
-      end
+    @applicable_document_codes ||= ApplicableDocumentCodesService.new.call
```

Comment:
You could inject the dependency to avoid having a hard dependency on `UserSession.get` - it would avoid needing to stub the `UserSession.get` method in specs?

`@applicable_document_codes ||= ApplicableDocumentCodesService.new(UserSession.get).call`

Reply:
Feels a bit weird injecting a global :thinking:

Reply:
I'll have a look in the follow on PR since I'm making changes to these tests